go-controller/pkg/ovn/pods.go

```diff
@@ -184,7 +186,25 @@ func getRoutesGatewayIP(pod *kapi.Pod, gatewayIPnet *net.IPNet) ([]util.PodRoute
 			routes = append(routes, route)
 		}
 	} else {
-		gatewayIP = gatewayIPnet.IP
+		if namespaceExternalGW == "" {
+			gatewayIP = gatewayIPnet.IP
+		} else {
+			for _, clusterSubnet := range config.Default.ClusterSubnets {
```

Comment:
@rcarrillocruz why is this block here, and not reusing the same code just above?